### ISSUE TITLE
[processor/routing] Deprecate component

### ISSUE THREAD

**Issue by djaglowski** [open-telemetry/opentelemetry-collector-contrib#19739]
### Component(s)

processor/routing

### Describe the issue you're reporting

This is a tracking issue relating to https://github.com/open-telemetry/opentelemetry-collector/issues/7370

The processor should be reimplemented as a connector. (See #19738) 

We should deprecate the processor once the connector is available.

**Comment by mwear** [comment]
If this is up for grabs, I can work on it.

**Comment by mwear** [comment]
I am volunteering to convert it to a connector first. Should I ask the same on #19738?

**Comment by djaglowski** [comment]
No, just wanted to make sure. I'll assign the connector issue to you. Thank you for helping with this!

**Comment by mwear** [comment]
No problem. Thanks for assigning!

**Comment by mx-psi** [comment]
Is there anything blocking this, or has it just not been taken up by anybody?

**Comment by djaglowski** [comment]
I believe the routing connector only matches the processor's OTTL abilities, which is limited to routing based on resource context. If I recall correctly, there was a discussion about this in a SIG and the decision was to not deprecate the processor until the connector's OTTL abilities could fully match the processor's non-OTTL implementation.

That said, I would not be opposed to deprecating the processor now and expecting that work continues on the connector/ottl to fill the gaps as necessary. Of course we should be confident that it is a full replacement before removing the component though.

**Comment by mx-psi** [comment]
If we confirm the eventual feature parity, I would vote for deprecating this component, so that we can make progress on open-telemetry/opentelemetry-collector/issues/7370. If the processor is here to stay, then we should think about a way to unblock open-telemetry/opentelemetry-collector/issues/7370 (an intermediate step could be to remove the method from the `component.Host` interface but still have the underlying struct implement it temporarily)

**Comment by mx-psi** [comment]
> Can you clarify what you mean here?

Mostly what you said above: if we confirm that the connector can be a full replacement of the processor, we can deprecate the processor now, even if the connector doesn't have all the features today.

**Comment by mwear** [comment]
Future feature parity is still on my todo list. I've been busy with component health and the health check extension. That said, the health related work is likely to continue for the foreseeable future. If any wants to own closing the feature gaps, I'd be happy to hand it off, but if nobody gets to it, I'll resume this work when my schedule frees up.

**Comment by djaglowski** [comment]
Closed by [[contrib] remove deprecated routing processor open-telemetry/opentelemetry-collector-releases#819](https://github.com/open-telemetry/opentelemetry-collector-releases/pull/819)

**Comment by atoulme** [comment]
@djaglowski should we track eventual removal from contrib?

**Comment by djaglowski** [comment]
Good point. I've reopened and we should close only when the codebase is removed.
